File: atcoder/abc323a_weak_beats.cpp
// Vicfred
// https://atcoder.jp/contests/abc323/tasks/abc323_a
// implementation
#include <iostream>

using namespace std;

int main() {
  string S;
  cin >> S;
  int N = S.size();
  bool flag = true;
  for (int i = 0; i < N; ++i) {
    if (S[i + 1] == '1' and i % 2 == 0) {
      flag = false;
      break;
    }
  }
  if (flag) {
    cout << "Yes" << endl;
  } else {
    cout << "No" << endl;
  }
  return 0;
}

File: atcoder/abc323a_weak_beats.hs
-- vicfred
-- https://atcoder.jp/contests/abc323/tasks/abc323_a
-- implementation
import Data.Bool (bool)

main :: IO ()
main = getLine >>= putStrLn .bool "No" "Yes"
                            .all (== True)
                            .map f
                            .(zip [1..])

f :: (Int, Char) -> Bool
f (n, c)
  | even n && c =='1' = False
  | otherwise = True
